[DataGrid] TypeScript error on 'GridToolbarFilterButtonProps' after upgrading @mui/material to v5.0.3

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

After upgrading the core to the latest version (5.0.3) I'm getting this typescript error running `tsc`:

```
$ tsc --project tsconfig.json && eslint --ignore-path .gitignore --fix --ext js,jsx,ts,tsx .

node_modules/@mui/x-data-grid-pro/x-grid.d.ts:2414:11 - error TS2430: Interface 'GridToolbarFilterButtonProps' incorrectly extends interface 'Omit<TooltipProps, "title" | "children">'.
  Types of property 'componentsProps' are incompatible.
    Type '{ button?: ButtonProps<"button", {}> | undefined; } | undefined' is not assignable to type '{ popper?: (PopperProps & TooltipComponentsPropsOverrides) | undefined; transition?: (TransitionProps & TooltipComponentsPropsOverrides) | undefined; tooltip?: (HTMLProps<...> & ... 1 more ... & TooltipComponentsPropsOverrides) | undefined; arrow?: (HTMLProps<...> & ... 1 more ... & TooltipComponentsPropsOverrides) ...'.
      Type '{ button?: ButtonProps<"button", {}> | undefined; }' has no properties in common with type '{ popper?: (PopperProps & TooltipComponentsPropsOverrides) | undefined; transition?: (TransitionProps & TooltipComponentsPropsOverrides) | undefined; tooltip?: (HTMLProps<...> & ... 1 more ... & TooltipComponentsPropsOverrides) | undefined; arrow?: (HTMLProps<...> & ... 1 more ... & TooltipComponentsPropsOverrides) ...'.

2414 interface GridToolbarFilterButtonProps extends Omit<TooltipProps, 'title' | 'children'> {
```

<img width="500" src="https://user-images.githubusercontent.com/42154031/136420155-39a6fc63-8e05-42f1-9cb5-633c2488513d.png" alt="image" style="max-width: 100%;">

### Expected behavior 🤔

The app build works 😅 

### Steps to reproduce 🕹

Steps:

1. update `@mui/material` to v5.0.3
2. run `tsc`
3. get the error in console


### Context 🔦

Relevant `package.json` content:

```
"@emotion/react": "^11.4.1",
"@emotion/styled": "^11.3.0",
"@mui/lab": "^5.0.0-alpha.50",
"@mui/material": "^5.0.3",
"@mui/styles": "^5.0.1",
"@mui/x-data-grid-pro": "5.0.0-beta.2",
"react": "^17.0.2",
"react-dom": "^17.0.2",
```

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```

  System:
    OS: macOS 11.6
  Binaries:
    Node: 16.9.1 - /opt/homebrew/bin/node
    Yarn: Not Found
    npm: 7.21.1 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: Not Found
    Edge: 94.0.992.37
    Firefox: 86.0.1
    Safari: 15.0
  npmPackages:
    @emotion/react: ^11.4.1 => 11.4.1 
    @emotion/styled: ^11.3.0 => 11.3.0 
    @mui/core:  5.0.0-alpha.50 
    @mui/lab: ^5.0.0-alpha.50 => 5.0.0-alpha.50 
    @mui/material: ^5.0.3 => 5.0.3 
    @mui/private-theming:  5.0.1 
    @mui/styled-engine:  5.0.1 
    @mui/styles: ^5.0.1 => 5.0.1 
    @mui/system:  5.0.3 
    @mui/types:  7.0.0 
    @mui/utils:  5.0.1 
    @mui/x-data-grid-pro: 5.0.0-beta.2 => 5.0.0-beta.2 
    @mui/x-license-pro:  5.0.0-beta.2 
    @types/react: ^17.0.27 => 17.0.27 
    react: ^17.0.2 => 17.0.2 
    react-dom: ^17.0.2 => 17.0.2 
    typescript: ^4.4.3 => 4.4.3 
```
</details>


### Order ID 💳 (optional)

29385

--- a/packages/grid/_modules_/grid/components/toolbar/GridToolbarFilterButton.tsx
+++ b/packages/grid/_modules_/grid/components/toolbar/GridToolbarFilterButton.tsx
@@ -36,9 +36,7 @@ export interface GridToolbarFilterButtonProps
    * The props used for each slot inside.
    * @default {}
    */
-  componentsProps?: ('componentsProps' extends keyof TooltipProps
-    ? Pick<TooltipProps, 'componentsProps'>
-    : {}) & { button?: ButtonProps };
+  componentsProps?: { button?: ButtonProps };
 }
 
 const GridToolbarFilterButton = React.forwardRef<HTMLButtonElement, GridToolbarFilterButtonProps>(

